drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -591,6 +596,22 @@ def collect_record_validations(user: nil, household_members: nil)
     end
   end
 
+  # @return <Hash{container_name:field_name=> FormItem }>
+  private def mapped_form_items
```

Comment:
I'm trying to present a sort of flexible solution here so that, in case there is ever another future use case where we want to know something about the form item while processing the submitted value, we can reuse this.

I briefly considered trying to combine this with `mapped_custom_form_fields` and `mapped_record_form_fields` so we don't have to construct so many maps: both for the sake of improving code readability and reducing iterations thru the form definition. That could look, for example, like removing those 2, checking against this single map of definition items, and determining by looking at each item whether it's custom or record, and processing accordingly. But I figured that that level of disruption to this battle-tested code is probably not in scope. Let me know what you think.